### PR TITLE
Add `FlatGraphEvaluator` test

### LINKED ISSUE #363: Test `FlatGraphEvaluator`
Add a simple test that verifies that `FlatGraphEvaluator::evaluate` agrees with `GraphEvaluator::evaluate` on random data or hard-coded edge cases.

### PR DESCRIPTION
**Pull request by davidnevadoc**
This PR adds a test to ensure the flattened graph evaluator matches the original `GraphEvaluator`.
In particular, it ensures the chunking, rotations (with and without overflow) and all individual calculations behave as expected.

Closes #363

### CODE REVIEW

File: proofs/src/plonk/evaluation.rs [hunk 1]
```diff
@@ -1312,3 +1312,258 @@ pub fn evaluate<F: Field, B: PolynomialRepresentation>(
...
+                    }
+                    acc
+                }
+            };
+        }
+        *im.last().expect("non-empty graph")
+    }
+
+    /// Assert `FlatGraphEvaluator::{evaluate, evaluate_chunk}` agrees with the
+    /// ref_evalute for `graph` for several BATCH sizes.
```
Comment (miguel-ambrona) [inline proofs/src/plonk/evaluation.rs]:
```suggestion
    /// ref_evaluate for `graph` for several BATCH sizes.
```

File: proofs/src/plonk/evaluation.rs [hunk 2]
```diff
@@ -1312,3 +1312,258 @@ pub fn evaluate<F: Field, B: PolynomialRepresentation>(
...
+        let dbl = g.add_calculation(Calculation::Double(f1));
+        let neg = g.add_calculation(Calculation::Negate(mul));
+        let wb = g.add_calculation(Calculation::Mul(add, ValueSource::Beta()));
+        let wt = g.add_calculation(Calculation::Add(wb, ValueSource::Theta()));
+        let wc = g.add_calculation(Calculation::Sub(wt, ValueSource::TrashChallenge()));
+
+        // Constant as a Calculation operand.
+        let with_const = g.add_calculation(Calculation::Add(neg, ValueSource::Constant(2)));
+
+        // Horner caluclation to compute linear combination on challenge Y of previous
```
Comment (miguel-ambrona) [inline proofs/src/plonk/evaluation.rs]:
```suggestion
        // Horner calculation to compute linear combination on challenge Y of previous
```

File: proofs/src/plonk/evaluation.rs [hunk 3]
```diff
@@ -1312,3 +1312,258 @@ pub fn evaluate<F: Field, B: PolynomialRepresentation>(
...
+        poly::{LagrangeCoeff, Polynomial, Rotation},
+    };
+
+    fn rand_fq(rng: &mut ChaCha20Rng) -> Fq {
+        Fq::from(rng.next_u64()).double() + Fq::from(rng.next_u64())
+    }
+
+    fn rand_poly(rng: &mut ChaCha20Rng, n: usize) -> Polynomial<Fq, LagrangeCoeff> {
+        Polynomial {
+            values: (0..n).map(|_| rand_fq(rng)).collect(),
```
Comment (miguel-ambrona) [inline proofs/src/plonk/evaluation.rs]: Why not `Fq::random(&mut rng)`?

**Review by miguel-ambrona — DISMISSED** [review]

**Review by miguel-ambrona — APPROVED** [review]
